jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/bigquery/BigQuerySqlDialectTest.java

```diff
@@ -0,0 +1,130 @@
+    }
+
+    @Test
+    void testGetLiteralCapabilitiesLiteral() {
+        assertThat(this.dialect.getCapabilities().getLiteralCapabilities(),
+                containsInAnyOrder(NULL, BOOL, DATE, TIMESTAMP, EXACTNUMERIC, STRING));
+    }
+
+    @Test
+    void testGetPredicateCapabilities() {
+        assertThat(this.dialect.getCapabilities().getPredicateCapabilities(),
+                containsInAnyOrder(AND, OR, NOT, EQUAL, NOTEQUAL, LESS, LESSEQUAL, LIKE,
+                        REGEXP_LIKE, BETWEEN, IN_CONSTLIST, IS_NULL, IS_NOT_NULL));
+    }
+
+    @Test
+    void testGetScalarFunctionCapabilities() {
+        assertThat(this.dialect.getCapabilities().getScalarFunctionCapabilities(),
+                containsInAnyOrder(ABS, ACOS, ASIN, ATAN, ATAN2, CEIL, COS, COSH, DEGREES, DIV, EXP,
+                        FLOOR, GREATEST, LEAST, LN, LOG, MOD, POWER, RAND, ROUND, SIGN, SIN, SINH,
+                        SQRT, TAN, TANH, TRUNC, COLOGNE_PHONETIC, CONCAT, INSERT, INSTR, LENGTH,
+                        LOWER, LPAD, LTRIM, REGEXP_REPLACE, REPEAT, REPLACE, REVERSE, RIGHT, RPAD,
+                        RTRIM, SOUNDEX, SPACE, SUBSTR, TRIM, UPPER, CURRENT_DATE, CURRENT_TIMESTAMP,
+                        DATE_TRUNC, DAY, EXTRACT, MINUTE, MONTH, SECOND, WEEK, YEAR, ST_X, ST_Y,
+                        ST_LENGTH, ST_NUMPOINTS, ST_AREA, ST_BOUNDARY, ST_CENTROID, ST_CONTAINS,
+                        ST_DIFFERENCE, ST_DIMENSION, ST_DISJOINT, ST_DISTANCE, ST_EQUALS,
+                        ST_INTERSECTION, ST_INTERSECTS, ST_ISEMPTY, ST_TOUCHES, ST_UNION, ST_WITHIN,
+                        CAST, TO_TIMESTAMP, BIT_AND, BIT_OR, BIT_XOR, CASE, HASH_MD5));
+    }
+
+    @Test
+    void testGetAggregateFunctionCapabilities() {
+        assertThat(this.dialect.getCapabilities().getAggregateFunctionCapabilities(),
+                containsInAnyOrder(COUNT, COUNT_STAR, COUNT_DISTINCT, SUM, SUM_DISTINCT, MIN, MAX,
+                        AVG, AVG_DISTINCT, FIRST_VALUE, LAST_VALUE, STDDEV, STDDEV_DISTINCT,
+                        STDDEV_POP, STDDEV_POP_DISTINCT, STDDEV_SAMP, STDDEV_SAMP_DISTINCT,
+                        VARIANCE, VARIANCE_DISTINCT, VAR_POP, VAR_POP_DISTINCT, VAR_SAMP,
+                        VAR_SAMP_DISTINCT, GROUP_CONCAT, APPROXIMATE_COUNT_DISTINCT));
+    }
+
+    @Test
+    void testSupportsJdbcCatalogs() {
+        assertThat(this.dialect.supportsJdbcCatalogs(),
+                equalTo(SqlDialect.StructureElementSupport.MULTIPLE));
+    }
+
+    @Test
+    void testSupportsJdbcSchemas() {
+        assertThat(this.dialect.supportsJdbcSchemas(),
+                equalTo(SqlDialect.StructureElementSupport.MULTIPLE));
+    }
+
+    @Test
+    void testGetDefaultNullSorting() {
+        assertThat(this.dialect.getDefaultNullSorting(),
+                equalTo(SqlDialect.NullSorting.NULLS_SORTED_AT_END));
+    }
+
+    @ValueSource(strings = { "ab:\'ab\'", "a'b:'a''b'", "a''b:'a''''b'", "'ab':'''ab'''"})
```

Comment:
This test is probably 100% duplicated code. Let's move that to a generic place.

Reply:
moved